File: services/internal-treasury/src/main/java/br/com/deltaglobalbank/internal_treasury/infrastructure/persistence/repositories/JpaInternalTransferenceRepository.java
package br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.repositories;

import br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.entities.InternalTransferenceEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.UUID;

public interface JpaInternalTransferenceRepository extends JpaRepository<InternalTransferenceEntity, UUID> {
    Page<InternalTransferenceEntity> findByStatus(String status, Pageable pageable);
}

File: services/internal-treasury/src/main/kotlin/br/com/deltaglobalbank/internal_treasury/infrastructure/persistence/repositories/JpaInternalTransferenceRepository.kt
package br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.repositories


import br.com.deltaglobalbank.internal_treasury.infrastructure.persistence.entities.InternalTransferenceEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Page
import java.util.UUID

interface JpaInternalTransferenceRepository : JpaRepository<InternalTransferenceEntity, UUID> {
    fun findByStatus(status: String, pageable: Pageable): Page<InternalTransferenceEntity>
}